src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -207,21 +227,26 @@ def set_options(content):
 
         self.model_config = ModelConfig(
             data_root=self.config_path,
-            joblist=self.site_config.get_installed_jobs(),
+            joblist=self.site_config.job_list,
             refcase=self.ecl_config.refcase,
             config_content=user_config_content,
         )
 
     # build configs from config dict
     def _alloc_from_dict(self, config_dict):
+        site_config_parser = ConfigParser()
+        init_site_config_parser(site_config_parser)
+        site_config_content = site_config_parser.parse(site_config_location())
         # treat the default config dir
         self.config_path = os.getcwd()
         if ConfigKeys.CONFIG_DIRECTORY in config_dict:
             self.config_path = config_dict[ConfigKeys.CONFIG_DIRECTORY]
         config_dict[ConfigKeys.CONFIG_DIRECTORY] = self.config_path
 
         self.subst_config = SubstConfig(config_dict=config_dict)
-        self.site_config = SiteConfig.from_config_dict(config_dict=config_dict)
+        self.site_config = SiteConfig.from_config_dict(
```

Comment:
Shouldn't the config dictionary be completely resolved when used to build the site config? Meaning it should have the keys defined in the site-config and user-config. And if that is possible to have then I am not really sure you would still need the `from_config_content` logic.

Reply:
I don't understand what you mean by resolved.
We have two paths, one for a dictionary, which is unused at the moment, and one for a config content, which is used.

Reply:
Don't stress about it, from the discussion during the standup this is just an intermediate step so probably it will be fixed in the next refactor.